src/libs/metadata_view.c

```diff
@@ -791,6 +807,115 @@ void connect_key_accels(dt_lib_module_t *self)
   dt_accel_connect_lib(self, "jump to film roll", closure);
 }
 
+static void _update_layout(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  for(int i = 0; i < md_size; i++)
+  {
+    const gboolean hidden = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    for(int j = 0; j < 2; j++)
+    {
+      GtkWidget *w = gtk_grid_get_child_at(GTK_GRID(d->grid), j, i);
+      gtk_widget_show_all(w);
```

Comment:
this show_all is probably not needed as you are dealing with single widgets (not containers) here?

Reply:
Yes, you're right.